### PR TITLE
Validate view_content type

### PR DESCRIPTION
**Pull request by MikeOver1**
Closes https://github.com/DIAGNijmegen/rse-grand-challenge-admin/issues/51

### CODE REVIEW

File: app/grandchallenge/hanging_protocols/forms.py [hunk 1]
```diff
@@ -88,6 +88,14 @@ def clean_json(self):
 class ViewContentMixin:
     def clean_view_content(self):
         mapping = self.cleaned_data["view_content"] or {}
+        if not isinstance(mapping, dict):
```
Comment (jmsmkn) [inline app/grandchallenge/hanging_protocols/forms.py:91]: Shouldn't we be checking against the schema server side, not just that it is an object?

File: app/grandchallenge/hanging_protocols/forms.py [hunk 2]
```diff
@@ -88,6 +88,14 @@ def clean_json(self):
...
     def clean_view_content(self):
         mapping = self.cleaned_data["view_content"] or {}
+        if not isinstance(mapping, dict):
+            self.add_error(
+                error=f"Value {str(mapping)} is not valid. "
+                "Should be of type `object`.",
+                field="view_content",
+            )
+            # return, because the ci check assumes the value is a dict
+            return mapping
```
Comment (jmsmkn) [inline app/grandchallenge/hanging_protocols/forms.py:98]: Why return? Shouldn't a Validation error be raised here? I thought `add_error` should only be used on the model clean method, not individual fields.

Comment (MikeOverkamp-diag) [reply app/grandchallenge/hanging_protocols/forms.py:98]: Looked it up and you're absolutely right. I'll update the form and the test.

File: app/grandchallenge/hanging_protocols/forms.py (hunk 1 above)
Comment (MikeOverkamp-diag) [reply app/grandchallenge/hanging_protocols/forms.py]: Sorry, completely missed this remark.

Comment (jmsmkn) [reply app/grandchallenge/hanging_protocols/forms.py]: Do you know why this validation wasn't happening in the first place? It is set on the model field right? So why is it invalid here? Do the model validations just happen later?

Comment (jmsmkn) [reply app/grandchallenge/hanging_protocols/forms.py]: > I think the model validation happens after the form validates

Can you confirm this?

Comment (MikeOverkamp-diag) [reply app/grandchallenge/hanging_protocols/forms.py]: It does, in ModelForms, model.full_clean() is called after the form's clean method.

**Review by jmsmkn — APPROVED** [review]

**Pull request merged**